### DISCUSSION TITLE
Why i cannot declaring  stackalloc MyStr[10]  when MyStr is 'ref struct'

### DISCUSSION THREAD

**Discussion by P9avel** [dotnet/runtime#94801] [General] [answered]
I am want putting all in stack. But my code not compiled. I dont undetstand why.

public ref struct MyStr
{
    public int I { get; set; }
}

internal class Program
{
    static void Main(string[] args)
    {
        Span<MyStr> arr = stackalloc MyStr[10];

        MyStr ms = new MyStr()
        {
            I = 123
        };

        Console.WriteLine("Hello, World!");
    }
}

Can you describe?

**Comment by gfoidl** [comment, marked as answer]
Because `MyStr` is used as type argument for `T` in `Span<T>`, and `ref struct`s are not allowed as type arguments.

You could
* `MyStr* a = stackalloc MyStr[10];`
* use `[InlineArray]` (.NET 8)

to work around this.
